v2/cmd/driver/main.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2021 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/kubeflow/pipelines/api/v2alpha1/go/pipelinespec"
+	"github.com/kubeflow/pipelines/v2/driver"
+	"github.com/kubeflow/pipelines/v2/metadata"
+)
+
+const (
+	driverTypeArg = "type"
+)
+
+var (
+	// inputs
+	driverType        = flag.String(driverTypeArg, "", "task driver type, one of ROOT_DAG, CONTAINER")
+	pipelineName      = flag.String("pipeline_name", "", "pipeline context name")
+	runID             = flag.String("run_id", "", "pipeline run uid")
+	componentSpecJson = flag.String("component", "{}", "component spec")
+	taskSpecJson      = flag.String("task", "{}", "task spec")
+	runtimeConfigJson = flag.String("runtime_config", "{}", "jobruntime config")
+
+	// container inputs
+	dagContextID   = flag.Int64("dag_context_id", 0, "DAG context ID")
```

Comment:
Is 0 a valid dag_context_id? Do we need to specify negative number here?

Reply:
No, not valid, it starts with 1.